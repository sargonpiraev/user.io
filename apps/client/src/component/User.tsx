import * as React from 'react'
import { Link } from 'react-router'
import { Item } from 'semantic-ui-react'
import { IUserState } from './../store/reducer'
import ProgressiveImage from 'react-progressive-image'

/* tslint:disable */
export const placeholder = 'data:image/svg+xml;base64,PHN2ZyBmaWxsPScjMDAwMDAwJyBoZWlnaHQ9JzQ4JyB2aWV3Qm94PScwIDAgMjQgMjQnIHdpZHRoPSc0OCcgeG1sbnM9J2h0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnJz48cGF0aCBkPSdNMCAwaDI0djI0SDB6JyBmaWxsPSdub25lJy8+PHBhdGggZD0nTTIxIDE5VjVjMC0xLjEtLjktMi0yLTJINWMtMS4xIDAtMiAuOS0yIDJ2MTRjMCAxLjEuOSAyIDIgMmgxNGMxLjEgMCAyLS45IDItMnpNOC41IDEzLjVsMi41IDMuMDFMMTQuNSAxMmw0LjUgNkg1bDMuNS00LjV6Jy8+PC9zdmc+'
/* tslint:enable */

export default ({ id, name, avatarUrl }: IUserState) => (
	<Item.Group className='user'>
		<Item>
			<ProgressiveImage src={ avatarUrl } placeholder={ placeholder }>
				{ (src: string) => <Item.Image size='mini' src={ src } /> }
			</ProgressiveImage>
			<Item.Content>
				<Item.Header as='h5'>
					<Link to={ `/user/${ id }` } >{ name }</Link>
				</Item.Header>
				<Item.Description>ID: { id }</Item.Description>
			</Item.Content>
		</Item>
	</Item.Group>
)
