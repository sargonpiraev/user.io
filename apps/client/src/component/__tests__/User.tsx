import 'core-js/es6/map'
import 'core-js/es6/set'
import 'raf/polyfill'

import * as React from 'react'
import * as renderer from 'react-test-renderer'
import User from './../User'

test('User', () => {
	const component = renderer.create(
		<User id={ 1 } name='John Doe' avatarUrl='https://ya.ru/favicon.ico' />
	)
	const tree = component.toJSON()
	expect(tree).toMatchSnapshot()
})
