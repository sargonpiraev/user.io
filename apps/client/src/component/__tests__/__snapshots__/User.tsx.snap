// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User 1`] = `
<div
  className="ui items user"
>
  <div
    className="item"
  >
    <div
      className="ui mini image"
      href={undefined}
    >
      <img
        src="https://ya.ru/favicon.ico"
      />
    </div>
    <div
      className="content"
    >
      <h5
        className="header"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          John Doe
        </a>
      </h5>
      <div
        className="description"
      >
        ID: 
        1
      </div>
    </div>
  </div>
</div>
`;
